src/agent_writer.cpp

```diff
@@ -0,0 +1,173 @@
+#include "agent_writer.h"
+#include <iostream>
```

Comment:
Could we please undo this change, and redo it with `git mv`. Otherwise the history isn't kept and I can't see changes.

Reply:
I tried to fix this, but github isn't showing things clearly.
The history correctly shows up in blame, but not in the file's commit logs or in the PR diff